rts/Map/SMF/ROAM/RoamMeshDrawer.cpp

```diff
@@ -188,17 +195,81 @@ void CRoamMeshDrawer::Update()
 
 	bool shadowPass = (cam->GetCamType() == CCamera::CAMTYPE_SHADOW);
 	bool tesselMesh = forceNextTesselation[shadowPass];
+	bool fullRetesselate = false;
 
 	auto& patches = patchMeshGrid[shadowPass];
 	auto& pvflags = patchVisFlags[shadowPass];
 
 	Patch::UpdateVisibility(cam, patches, numPatchesX);
 
-#define RETESSELLATE_MODE 1
+    int numPatchesVisible = 0;
+    int numPatchesEnterVisibility = 0;
+    int numPatchesEnteredSkipped = 0;
+    int numPatchesExitVisibility = 0;
+    int numPatchesRetesselated = 0;    
+    bool patchesToTesselate[numPatchesX * numPatchesY];
```

Comment:
this isn't standard C++, you can't create an array of dynamic size like this.
consider:
1) using C arrays and dynamically allocating them
2) using STL constructs like vector (vector of bool can be weird), unique_ptr or deque

Reply:
vector is the best way I think, might be worth to reuse it so you don't reallocate it each time